django/db/backends/base/features.py

```diff
@@ -128,38 +128,20 @@ class BaseDatabaseFeatures:
     # Can the backend introspect an AutoField, instead of an IntegerField?
     can_introspect_autofield = False
 
-    # Can the backend introspect a BigIntegerField, instead of an IntegerField?
-    can_introspect_big_integer_field = True
-
-    # Can the backend introspect an BinaryField, instead of an TextField?
-    can_introspect_binary_field = True
-
-    # Can the backend introspect an DecimalField, instead of an FloatField?
-    can_introspect_decimal_field = True
-
-    # Can the backend introspect a DurationField, instead of a BigIntegerField?
-    can_introspect_duration_field = True
-
-    # Can the backend introspect an IPAddressField, instead of an CharField?
-    can_introspect_ip_address_field = False
-
-    # Can the backend introspect a PositiveIntegerField, instead of an IntegerField?
-    can_introspect_positive_integer_field = False
-
-    # Can the backend introspect a SmallIntegerField, instead of an IntegerField?
-    can_introspect_small_integer_field = False
-
-    # Can the backend introspect a TimeField, instead of a DateTimeField?
-    can_introspect_time_field = True
-
-    # Some backends may not be able to differentiate BigAutoField or
-    # SmallAutoField from other fields such as AutoField.
-    introspected_big_auto_field_type = 'BigAutoField'
-    introspected_small_auto_field_type = 'SmallAutoField'
-
-    # Some backends may not be able to differentiate BooleanField from other
-    # fields such as IntegerField.
-    introspected_boolean_field_type = 'BooleanField'
+    # Map fields which some backends may not be able to differentiate to the
+    # field it's introspected as.
+    introspected_field_types = {
+        'BigAutoField': 'BigAutoField',
+        'BigIntegerField': 'BigIntegerField',
+        'BinaryField': 'BinaryField',
+        'BooleanField': 'BooleanField',
+        'DurationField': 'DurationField',
+        'IPAddressField': 'CharField',
+        'PositiveIntegerField': 'IntegerField',
```

Comment:
I think all the defaults here should be mappings to the same type (e.g. `'PositiveIntegerField': 'PositiveIntegerField'`), so that backends have to specify the types they introspect differently.